src/transformers/tokenization_utils_base.py

```diff
@@ -3011,9 +3012,27 @@ def truncate_sequences(
                 logger.error(
                     f"We need to remove {num_tokens_to_remove} to truncate the input"
                     f"but the first sequence has a length {len(ids)}. "
-                    f"Please select another truncation strategy than {truncation_strategy}, "
-                    f"for instance 'longest_first' or 'only_second'."
+                    + (
+                        (
+                            f"Please select another truncation strategy than {truncation_strategy}, "
+                            f"for instance 'longest_first' or 'only_second'."
+                        )
+                        if truncation_strategy == TruncationStrategy.ONLY_FIRST
+                        else ""
+                    )
                 )
```

Comment:
This is unreadable, please define the error message outside of the logger to have better formatting.